Spatial_searching/include/CGAL/Kd_tree.h

```diff
@@ -336,7 +448,50 @@ class Kd_tree {
                                "Parallel_tag is enabled but TBB is unavailable.");
 #endif
 
-    Point_container c(dim_, data.begin(), data.end(),traits_);
+    std::vector< std::vector<FTP> > references;
+    long start = -1, end = -1;
+    if (m_create_balanced_tree) {
+
+      references.resize(dim_, std::vector<FTP>(data.size()));
+      std::vector<std::size_t> ref_end(references.size());
```

Comment:
`references,size()` is `dim_`.

Reply:
This part has changed after we did a recent speedup.